frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteFileUtils.java

```diff
@@ -30,19 +36,30 @@ public static String getSampleMacheteFileContents() {
 
   @UIThreadUnsafe
   public static List<String> getBranchNamesForPsiMacheteFile(PsiFile psiFile) {
-    val project = psiFile.getProject();
-
-    val gitRepository = List.ofAll(GitRepositoryManager.getInstance(project).getRepositories())
-        .find(repository -> {
-          val macheteFile = repository.getMacheteFile();
-          return macheteFile != null && macheteFile.equals(psiFile.getVirtualFile());
-        });
+    val gitRepository = findGitRepositoryForPsiMacheteFile(psiFile);
 
-    if (gitRepository.isEmpty()) {
+    if (gitRepository == null) {
       return List.empty();
     }
 
-    return List.ofAll(gitRepository.get().getInfo().getLocalBranchesWithHashes().keySet())
+    return List.ofAll(gitRepository.getInfo().getLocalBranchesWithHashes().keySet())
         .map(localBranch -> localBranch.getName());
   }
+
+  @Nullable
+  @UIThreadUnsafe
+  public static GitRepository findGitRepositoryForPsiMacheteFile(PsiFile psiFile) {
```

Comment:
Nit:
```suggestion
  @UIThreadUnsafe
  public static @Nullable GitRepository findGitRepositoryForPsiMacheteFile(PsiFile psiFile) {
```
As `@Nullable` is about the type, not the method